src/pcapi/core/bookings/validation.py

```diff
@@ -96,6 +96,17 @@ def check_offerer_can_cancel_booking(booking: Booking) -> None:
         raise forbidden
 
 
+def check_system_can_cancel_expired_booking(booking: Booking) -> None:
+    if not booking.stock.offer.offerType["canExpire"]:
+        raise exceptions.DoesNotExpire(
+            "Booking %s is of type %s, which does not expire", booking, booking.stock.offer.type
```

Comment:
Je ne suis pas sûr qu'on puisse passer des arguments à ton exception comme ça. Testé en local sur un exemple plus simple :

```python
try:
    raise Exception("foo %s", 1)
except Exception as exc:
    print(str(exc))
```

... donne la sortie suivante :

```Exception: ('foo %s', 1)```

Suggestion :

```suggestion
            "Booking %s is of type %s, which does not expire % (booking, booking.stock.offer.type)
```

... idem pour les autres exceptions plus bas.